manifests/tools.pp

```diff
@@ -1,8 +1,20 @@
-# Class to ensure qpid tools package
-class qpid::tools {
+# Class to manage qpid tools package
+#
+# @param ensure
+#  Specify 'present' to install qpid-tools package or 'absent' to remove the package and any dependencies
+#
+class qpid::tools(
+  Enum['present', 'absent'] $ensure = 'present',
+) {
+
+  if $ensure == 'absent' {
+    $_package_ensure = 'purged'
+  } else {
+    $_package_ensure = $ensure
```

Comment:
Should this get set to `installed` instead of present? I'm not clear from the documentation on the package type what the difference is between those two.

Reply:
AFAICT, the only difference is style. Older versions of the docs explicitly state that these are equivalent; for some reason the newer docs don't say anything about it.

Other Resource types also seem have have some allowed values for `ensure` that are aliases. I think `file { $filename: ensure => {'file','present'}` are equivalent, but it doesn't make sense to have any generic alias of `file { $path: ensure => 'directory' }`.

The Puppet 7 style guide at https://puppet.com/docs/puppet/7.5/style_guide.html has an implicit convention of using  `package { $name: ensure => 'present' }` throughout, without specifically calling that out as something one should do.

Reply:
Related style thought:

I doubt it would make much difference here, but as a general practice, is it actually better to specifically pass the string 'present' rather than the value of `$ensure`? I suppose it is easier to read that way, and allows a bit more separation of the interface for `qpid::tools` vs. the values it passes to the package resource.

Reply:
Another point is, `ensure => 'installed'` is a little more readable/greppable than `ensure => present`. I really don't have a strong preference either way. If you do, let me know and I'll make the quick change.

Otherwise, is there anything else you or @ekohl would like to review further with this PR or should we move forward?

Reply:
I've always understood `installed` was just easier to read. A quick look at the source tells me they're really an alias as @wbclark said.

https://github.com/puppetlabs/puppet/blob/b9240c317780df0359eb2451db0a4cb0177a353a/lib/puppet/type/package.rb#L113-L114

Which makes sense: installed means it's present and present means it's installed. If we compare it to file type, there isn't a symmetrical relationship. You can say that directory implies it's present but just because it's present doesn't mean it's a directory. That's why it can be good to avoid `present` on the file type: it's ambiguous. That's why https://github.com/voxpupuli/puppet-lint-file_ensure-check exists.